components/leaflet-maps/ClustersMap.vue

```diff
@@ -195,7 +195,53 @@
        })
 
        return optionalsWithValue
-     }
+     },
+     getFormatDate: function (date,precision){
+
+      let [day, month, year] = date.split('/');
+      day = Number(day);
+      month = Number(month)
+      year = Number(year);
+      let currentDate = new Date(year, month - 1, day);
+      
+      
+      //the precision of the dates can either be or 9 11(full day/month/year),10(month/year),9(year)
+      
+      // If we don't know the day, month, and year, then wikidata precision will be less than 11 and the dates don't need Julian conversion
+      if (precision === '10') {
+          // Then return the date with month and year only
+          let dateString = `${(currentDate.getMonth() + 1).toString().padStart(2, '0')}/` + `${currentDate.getFullYear()}`;
+          return dateString;
+      }  
+          // If only year is known
+      if (precision === '9')  {
+          // Then return the date with year only
+          let dateString = `${currentDate.getFullYear()}`;
+          return dateString;
+      }
+        
+      
+
+      //if we have the full date (day,month,year) them the date needs to be converted into Julian
+      const cutoffDate = new Date(1700,3,1) // the date where the difference between julian and gregorian cjanges from 10 to 11
+      let earlierDate = new Date(currentDate);
+
+      // JULIAN CONVERSION - https://en.wikipedia.org/wiki/Conversion_between_Julian_and_Gregorian_calendars
+      if (currentDate < cutoffDate) {
+        earlierDate.setDate(currentDate.getDate()-10);
+      }
+      else {
+        earlierDate.setDate(currentDate.getDate()-11);
+      }
+
+      // covert back to string
+      const earlierDateString =
```

Comment:
you could replace this section of code with the following:

```
return earlierDate.toLocaleDateString('en-gb');
```

This should make formatting back to strings a bit easier.

Reply:
this returns the date in the format "Thursday, 20 December 2012". I'm not sure if we want this, what do you think is better - "Thursday, 20 December 2012" or 20/12/2012?

Reply:
We can control the format. Without specifying options as the second argument, it should default to 20/12/2012 for example.

However, as discussed, I think having some dates that just show month and year and others that show day month and year is a bit jarring. I think it looks better as below:

![image](https://github.com/uoe-dlam/witches/assets/1457058/d7b8789f-da7d-4d68-b80f-a920a0a32d04)

This can be achieved using something like:

```
 return julianDate.toLocaleDateString("en-GB", {
      year: "numeric",
      month: "short",
      day: "numeric",
    });

```

Cheers!

Reply:
Dates changes so marking as resolved. Thanks Ruby